news/_posts/2020-06-18-about-bc-breaks.md

```diff
@@ -0,0 +1,398 @@
+
+### How PrestaShop open source project maintainers handle this type of BC break
+
+We use the deprecations system to warn people as much as possible about incoming BC breaks.
+
+## HTML Markup BC break
+
+Since PrestaShop is a web project, we have dozen of HTML templates inside the codebase that evolve in time. These templates have a structure and markers such as IDs and classes, that evolve too. These evolutions can introduce a BC break.
+
+For example a lot of JavaScript code, such as jQuery, relies on HTML IDs and classes to select DOM elements to modify. This means that modifying these markers or removing them... break the JavaScript behavior relying on them !
+
+As a consequence, in order to comply with SemVer on the PrestaShop project, maintainers also make sure to keep HTML markup backward compatible.
+
+**Hard markup BC break example:**
+```html
+<div id="my-id"><a id="link" href="#anchor">Link</a></div>
+```
+becomes simply
+```html
+<div id="my-id"><a id="anchor" href="#anchor">Link</a></div>
+```
+In the above example, the ID "link" has been renamed to "anchor". Any external code that relied on this selector will now fail.
+
+**Soft BC break example:**
+```html
+<div id="my-id"><a id="link" href="#anchor">Link</a></div>
+```
+becomes
+```html
+<div id="link"><a href="#anchor">Link</a></div>
+```
+In the above example, the ID "link" was moved from the `<a>` to the `<div>`. Any external code that relied on this selector might fail because of this structural change.
+
+Example of such BC break: recently we had to modify some CSS code in order to fix a bug for mobile display, and this could not be done [without introducing a breaking change into the style in PR 17438](https://github.com/PrestaShop/PrestaShop/pull/17438).
+
+### How PrestaShop open source project maintainers handle this type of BC break
+
+We try not to introduce such changes as there is no such thing as deprecation for markup. If however we are forced to do so, we try to inform users of the change into the Release Note.
+
+## Dependency BC break
+
+Similarly to other projects of its size, PrestaShop "is built on the shoulders of giants" and use a set of PHP and JavaScript dependencies. These dependencies provide great software assets that are being used to implement all of PrestaShop behaviors and are shipped with PrestaShop codebase inside the final ZIP archive.
+For example PrestaShop uses the popular [PHP HTTP client Guzzle](http://docs.guzzlephp.org/en/stable/). This means that module developers can rely on it too in their implementations.
+And consequently removing or modifying these dependencies can introduce a BC break too.
+
+This type of BC break is especially hard to deal with because we have no control on these packages. Sometimes, in order to avoid introduce a dependency BC break, [we decided to maintain a package ourselves](https://github.com/PrestaShop/CsaGuzzleBundle) rather than using the latest version because of the hard BC break it would introduce.
```

Comment:
```suggestion
This type of BC break is especially hard to deal with because we have no control on these packages. Sometimes, in order to avoid introducing a dependency BC break, [we decided to maintain a package ourselves](https://github.com/PrestaShop/CsaGuzzleBundle) rather than using the latest version because of the hard BC break it would introduce.
```